pkg/disk/controllerserver.go

```diff
@@ -653,10 +668,41 @@ func (cs *controllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
 	}
 	createSnapshotRequest.Tag = &snapshotTags
 
+	pvcName, preRule, postRule, err := CheckAppConsistent(volumeSnapshotName, volumeSnapshotNamespace)
+	var podName string
+	log.Infof("CreateSnapshot:: appconsistent status pvcName: %s namespace: %v", pvcName, volumeSnapshotNamespace)
+	if pvcName != "" {
+		pods, err := GetPodsByPvc(volumeSnapshotNamespace, pvcName)
+		if err != nil {
+			log.Errorf("CreateSnapshot:: get pods by pvc error: %v, pvcName: %v, volumeSnapshotNamespace: %v", err, pvcName, volumeSnapshotNamespace)
+		}
+		if len(pods) == 1 {
+			podName = pods[0].Name
+		} else {
+			log.Errorf("CreateSnapshot:: get pods count: %v by pvc: %v which is not support for now", len(pods), pvcName)
+		}
+	}
+	if preRule.Spec.ContainerName != postRule.Spec.ContainerName {
+		log.Errorf("CreateSnapshot:: container name in preRule & postRule are must be same")
+		podName = ""
+	}
+	if err == nil && preRule.Name != "" && postRule.Name != "" && podName != "" {
+		command := utils.NewPodCommands(volumeSnapshotNamespace, podName, preRule.Spec.ContainerName, GlobalConfigVar.ClientSet, GlobalConfigVar.kubeClientConfig)
+		err = command.ApplyPreCommands(preRule)
```

Comment:
CreateSnapshot should be took place between ApplyPreCommands and ApplyPostCommands. Immediate ApplyPostCommands here will lead to release IO lock and snapshot will go to in-consistent condition.